pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -1,32 +1,42 @@
-hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
-# Intentionally word-split in case 'hardeningEnable' is defined in Nix.
-hardeningFlags+=(${hardeningEnable[@]})
+allHardeningFlags=(fortify stackprotector pie pic strictoverflow format relro bindnow)
 hardeningCFlags=()
 hardeningLDFlags=()
 
-declare -A hardeningDisableMap
+declare -A hardeningDisableMap=()
+declare -A hardeningEnableMap=()
 
-# Intentionally word-split in case 'hardeningDisable' is defined in Nix. The
+# Create table of unsupported flags for this toolchain.
+for flag in @hardening_unsupported_flags@; do
```

Comment:
Hmm, upon further reading, maybe something like:
```bash
if [[ -n "$NIX_DEBUG" ]]; then
    declare -A hardeningDisableMap=()
fi
for flag in @hardening_unsupported_flags@; do
    [[ -n ${hardeningEnableMap[$flag]} ]] | continue
    if [[ -n "$NIX_DEBUG" ]]; then
        hardeningEnableMap[$flag]=1
    fi
    unset hardeningEnableMap[$flag]
done
if [[ -n "$NIX_DEBUG" ]]; then
    // print disabled flags as before
fi
```
Then it is clear that all the logic follows from the whitelist, and the disable map is only used for debugging.